src/utils/specific_label_utils.py

```diff
@@ -36,12 +36,12 @@ def differentiate_errors_in_string_column(data_column: pd.Series, generic_labele
             typo_word_map[word] = ErrorType.TYPO.value
         elif is_key_error(word, correct_words_list):
             typo_word_map[word] = ErrorType.TYPO.value
-        elif is_deletion(word, correct_words_list):
-            typo_word_map[word] = ErrorType.TYPO.value
         elif is_insertion_or_replication(word, correct_words_list):
             typo_word_map[word] = ErrorType.TYPO.value
         elif has_linguistic_misspelling_pattern(word, correct_words_list):
             typo_word_map[word] = ErrorType.MISSPELLING.value
+        elif is_deletion(word, correct_words_list):
+            typo_word_map[word] = ErrorType.TYPO.value
         else:
             typo_word_map[word] = ErrorType.OCR.value
```

Comment:
Not specifically related to this PR, but do we want to fall back to OCR? Or skip this?